models.py

```diff
@@ -559,11 +560,26 @@ def _urls_and_domains(self, auth_entity, user_url):
     for i, url in enumerate(candidates):
       final, domain, ok = util.get_webmention_target(url, resolve=i < MAX_AUTHOR_URLS)
       if ok:
-        if util.schemeless(final.lower()).startswith(util.schemeless(url.lower())):
+        current_url = ''
+        final = final.lower()
+        # Ignore redirects when fetching the root url, they are dealt with next.
+        if util.schemeless(final) == util.schemeless(url.lower()):
```

Comment:
we should also gate on `i < MAX_AUTHOR_URLS`, like with `resolve=` above, to limit the number of different root URLs we fetch. feel free to put it in a temporary variable if you want, e.g. `resolve`.

Reply:
yes should've picked that up too. I just added it to `if match:`, do you want that to be `if match and resolve:`?